stackrox/stackrox
# ROX-17011: Refactor compliance to enable local-compliance

## Description

This PR extracts all runtime dependencies of compliance into interfaces (refactoring part), so that compliance can now easily change behavior of:
1. ~Logging~
2. Generating Node-Inventories
3. Getting the name of the node it runs on
4. Handling the ACK/NACK messages from Sensor

The behavior of the Compliance component has not been changed in this PR.

This PR adds `local-compliance` that can be run locally and connect to the real Sensor (more info about this in the readme). The refactoring part was necessary to for `local-compliance` to happen. 

This PR is a prerequisite to enable Sensor integration tests in the scope defined in [ROX-16861](https://issues.redhat.com/browse/ROX-16861) 

## Checklist


### N/A

## Testing Performed

- Run local-compliance as documented in the readme
- Change compliance image to image from this branch on running cluster and confirm that it still runs as expected.



Based on a work started in branch `jschnath/ROX-17011_Investigate_the_viability_of_a_local_compliance`. Once merged, this branch can be deleted.
